test/checks/pagination_size_test.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+require "test_helper"
+
+module ThemeCheck
+  class PaginationSizeTest < Minitest::Test
+    def test_no_offense_for_good_behaviour
+      offenses = analyze_theme(
+        PaginationSize.new(max_size: 50),
+        "templates/index.liquid" => <<~END,
+          {% paginate products by 50 %}
+          {% endpaginate %}
+
+          {%- paginate products by 50 -%}
+          {%- endpaginate -%}
+
+          <!-- setting size -->
+          {%- paginate collection.products by section.settings.products_per_page -%}
+          {% endpaginate %}
+          {% schema %}
+            {
+                "name": "test",
+                "settings": [
+                    {
+                        "type": "number",
+                        "id": "products_per_page",
+                        "label": "Products per Page",
+                        "default": 12
+                    }
+                ]
+            }
+            {% endschema %}
+        END
+      )
+      assert_offenses("", offenses)
+    end
+
+    def test_flag_use_of_size_greater_than_limit
+      offenses = analyze_theme(
+        PaginationSize.new(max_size: 50),
+        "templates/index.liquid" => <<~END,
+          {%- paginate collection.products by 999 -%}
+          {%- endpaginate -%}
+          {% paginate collection.products by 999 %}
+          {% endpaginate %}
+        END
+      )
+      assert_offenses(<<~END, offenses)
+        Use a smaller pagination size at templates/index.liquid:1
+        Use a smaller pagination size at templates/index.liquid:3
+      END
+    end
+
+    def test_flag_use_of_setting_value
+      offenses = analyze_theme(
+        PaginationSize.new(max_size: 50),
+        "templates/index.liquid" => <<~END,
+          <!-- setting size -->
+          {%- paginate collection.products by section.settings.products_per_page -%}
+          {% endpaginate %}
```

Comment:
curious, it seems that you use `{% ... %}` and `{%- ... -%}` with/without the dashes back and forth in these tests - is there a reason for it?  I'm totally new to liquid, but i know it's related to whitespace.  I'm just wondering if this is an important thing to test here?

Reply:
These are just variations of the tag in Liquid, so I'm making sure that each variation is caught by the test. This is similar to how other tests check.